sdk/identity/azure-identity/azure/identity/_credentials/imds.py

```diff
@@ -39,6 +39,21 @@ def _get_request(scope: str, identity_config: Dict) -> HttpRequest:
     return request
 
 
+def _check_forbidden_response(ex: HttpResponseError) -> None:
+    """Special case handling for Docker Desktop.
+
+    Docker Desktop proxies all HTTP traffic, and if the IMDS endpoint is unreachable, it
+    responds with a 403 with a message that contains "A socket operation was attempted to an unreachable network".
+
+    :param ~azure.core.exceptions.HttpResponseError ex: The exception raised by the request
+    :raises ~azure.core.exceptions.CredentialUnavailableError: When the IMDS endpoint is unreachable
+    """
+    if ex.status_code == 403:
+        if ex.message and "A socket operation was attempted to an unreachable network" in ex.message:
```

Comment:
Is the error message always in English? Or we can want to handle the ones in English? :)

Reply:
Hah, I'm not too sure. I think just Engish only is fine, and aligns with what .NET and Go did for this edge-case handling.